@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -1667,4 +1673,133 @@ export class TileGeometryCreator {
+            topPlaneDistance.copy(bottomNormal).multiplyScalar(tile.boundingBox.extents.y);
+            bottomDistance.copy(topPlaneNormal).multiplyScalar(tile.boundingBox.extents.y);
+        }
+        // If Spherical, we need to compute the planes which intersection cut the earth on the shape
+        // of our tile. We cannot use the tiles' tangent-space-aligned boundingBoxes since they
+        // overlap.
+        else {
+            // Compute the world position of all the relevant tile points we will use to compute
+            // the clipping planes' plane equations.
+            const eastPoint = new THREE.Vector3();
+            const westPoint = new THREE.Vector3();
+            const northPoint = new THREE.Vector3();
+            const southPoint = new THREE.Vector3();
+            const northEastPoint = new THREE.Vector3();
+            const southWestPoint = new THREE.Vector3();
+            const northWestPoint = new THREE.Vector3();
+            const southEastPoint = new THREE.Vector3();
+            tile.projection.projectPoint(
+                new GeoCoordinates(tile.geoBox.center.latitude, tile.geoBox.east),
+                eastPoint
+            );
+            tile.projection.projectPoint(
+                new GeoCoordinates(tile.geoBox.center.latitude, tile.geoBox.west),
+                westPoint
+            );
+            tile.projection.projectPoint(
+                new GeoCoordinates(tile.geoBox.north, tile.geoBox.center.longitude),
+                northPoint
+            );
+            tile.projection.projectPoint(
+                new GeoCoordinates(tile.geoBox.south, tile.geoBox.center.longitude),
+                southPoint
+            );
+            tile.projection.projectPoint(tile.geoBox.northEast, northEastPoint);
+            tile.projection.projectPoint(tile.geoBox.southWest, southWestPoint);
+            tile.projection.projectPoint(
+                new GeoCoordinates(tile.geoBox.north, tile.geoBox.west),
+                northWestPoint
+            );
+            tile.projection.projectPoint(
+                new GeoCoordinates(tile.geoBox.south, tile.geoBox.east),
+                southEastPoint
+            );
+
+            // Set the plane distance to the difference between the plane's center and the tile's
+            // center in world space.
+            rightPlaneDistance.copy(eastPoint).sub(tile.center);
+            leftPlaneDistance.copy(westPoint).sub(tile.center);
+            topPlaneDistance.copy(northPoint).sub(tile.center);
+            bottomDistance.copy(southPoint).sub(tile.center);
+
+            // Compute the plane equation from the three relevant coplanar points for each plane (in
+            // counter-clockwise order), and extract their normals.
+            rightPlaneNormal.copy(
+                new THREE.Plane().setFromCoplanarPoints(southEastPoint, eastPoint, northEastPoint)
```

Comment:
can you prove this works? This would be the average normal, but wouldn't the normal at the edges of the tile's border be necessary?